app/models/invoice.rb

```diff
@@ -3,6 +3,19 @@ class Invoice < ApplicationRecord
   belongs_to :merchant
 
   has_many :transactions
+  has_many :invoice_items
 
   validates :status, presence: true
+
+  def total
+    invoice_items.reduce(0) do |total, item|
+      total += (item.unit_price * item.quantity)
+    end
+  end
+
+  def successful?
```

Comment:
Depends on what you mean. I like the idea of having some of this in separate methods that can then be reused, but I think it's possible to do that (maybe using a scope?) while still using only ActiveRecord.